### ISSUE TITLE
Assertion failed: (checkVariableImport(Index, ImportLists, ExportLists)), function ComputeCrossModuleImport

### ISSUE THREAD

**Issue by int3** [llvm/llvm-project#62468]
I ran into this assertion error in LTO code while working on `ld64.lld`. I was able to repro it on `ld.lld` too, so it seems like a general LTO issue. I am not super familiar with how LTO's FunctionImport is supposed to work, so I would appreciate help here. I've gotten it down to a simple repro (https://github.com/llvm/llvm-project/files/11362003/assert.repro.tar.gz) of two files:

#### foo.ll
```
target datalayout = "e-m:o-i64:64-i128:128-n32:64-S128"

@qux = linkonce_odr hidden global i32 0

define linkonce_odr hidden void @foo() {
  call void @bar()
  ret void
}

declare hidden void @bar()
```

#### bar.ll
```
target datalayout = "e-m:o-i64:64-i128:128-n32:64-S128"

@qux = linkonce_odr hidden global i32 0

define hidden void @bar() {
  %1 = load i32, i32* @qux, align 8
  ret void
}
```

`@qux` is marked read-only, and it is only present in the ExportLists but not the ImportLists, triggering the assertion. Disabling the assertion doesn't seem to result in any runtime issues though. Perhaps the assertion itself isn't correct? cc @eleviant and @teresajohnson --  it was introduced in https://reviews.llvm.org/D70128.

**Comment by teresajohnson** [comment]
The structure of how variable importing is done changed just this past week (D149298). Do you know or can you see if the assert happens prior to that change, committed Thu?

**Comment by int3** [comment]
Yep, my repro triggers the assert both before and after that change.

**Comment by teresajohnson** [comment]
Ok thanks for confirming. I will investigate more this week. Looking at the code and your test case, I suspect the assert might be too tight. I guess we are exporting it because of the reference in bar which is presumably being imported into foo.ll. But given the presumably prevailing linkonce_odr def of qux already in foo.ll we don't need to import it.

**Comment by teresajohnson** [comment]
Mailed you D149630 to prevent the assert in this case.
